server/world/recipes/recipe.go

```diff
@@ -0,0 +1,34 @@
+package recipes
+
+import "github.com/df-mc/dragonfly/server/item"
+
+// Recipe is embedded by all recipe types.
+type Recipe interface {
+	__()
+}
+
+// ShapelessRecipe is a recipe that has no particular shape.
+type ShapelessRecipe struct {
+	Recipe
```

Comment:
What's the purpose in embedding the `Recipe` interface?

Reply:
I embedded it here so I could have a generic form of "recipe" that other portions of the code could use, similar to action.go